actors/runtime/src/main/java/cloud/orbit/actors/runtime/DefaultLocalObjectsCleaner.java

```diff
@@ -166,16 +167,39 @@ private Task cleanupActors(final boolean shutdownAll)
 
     private boolean shouldRemove(final ActorBaseEntry<?> actorEntry, final Set<ActorBaseEntry<?>> toRemove)
     {
+        final Class<?> interfaceClass = RemoteReference.getInterfaceClass(actorEntry.getRemoteReference());
         // Make sure it isn't tagged NeverDeactivate
-        if (RemoteReference.getInterfaceClass(actorEntry.getRemoteReference()).isAnnotationPresent(NeverDeactivate.class))
+        if (interfaceClass.isAnnotationPresent(NeverDeactivate.class))
         {
             return false;
         }
-        // Check for timeout
-        boolean shouldRemove = clock.millis() - actorEntry.getLastAccess() > defaultActorTTL;
+
+        // Check for ttl override
+        if(interfaceClass.isAnnotationPresent(TimeToLive.class))
+        {
+            final TimeToLive customTtl = interfaceClass.getAnnotation(TimeToLive.class);
```

Comment:
Better to use our annotation cache here.

Reply:
I thought we decided we didn't need to use the annotation cache in these cases because the JVM cached? The annotation cache was for cases when we were inspecting methods, or am I misremebering?

Reply:
Think you're right.